api/handlers.go

```diff
@@ -0,0 +1,111 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/NebulousLabs/skynet-accounts/user"
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+
+	"github.com/julienschmidt/httprouter"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+// userHandlerGET returns information about an existing user.
+func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	// TODO Limit to the user themselves.
+	if err := req.ParseForm(); err != nil {
+		WriteError(w, errors.New("Failed to parse parameters."), http.StatusBadRequest)
+	}
+	email := req.Form.Get("email")
+	if email == "" {
+		WriteError(w, errors.New("No email provided."), http.StatusBadRequest)
+		return
+	}
+	users, err := api.DB.UserFindAllByField(req.Context(), "email", email)
+	if err != nil && err != database.ErrUserNotFound {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	if errors.Contains(err, database.ErrUserNotFound) || len(users) == 0 {
+		WriteError(w, database.ErrUserNotFound, http.StatusNotFound)
+		return
+	}
+	WriteJSON(w, users[0])
+}
+
+// userHandlerPOST creates a new user.
+func (api *API) userHandlerPOST(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	if err := req.ParseMultipartForm(64 * 1_000_000); err != nil {
+		WriteError(w, errors.New("Failed to parse multipart parameters."), http.StatusBadRequest)
+		return
+	}
+	email := (user.Email)(req.PostFormValue("email"))
+	if !email.Validate() {
+		WriteError(w, user.ErrInvalidEmail, http.StatusBadRequest)
+		return
+	}
+	user := &user.User{
+		FirstName: req.PostFormValue("firstName"),
+		LastName:  req.PostFormValue("lastName"),
+		Email:     email,
+	}
+	ins, err := api.DB.UserSave(req.Context(), user)
```

Comment:
Same here. I'd make this a `UserCreate` method to hide the database code within the database package. That would also make sure that the user's email is unique.

Reply:
Yes, will do. I left this as `save` because it satisfies both `create` and `update` but it will be messy in the long run.